docs/java-helloworld.md

```diff
@@ -0,0 +1,145 @@
+
+<!--SNIPSTART java-samples-logback-dependency-configuration-->
+<!--SNIPEND-->
+
+## Build "Hello World!" app
+
+Now we are ready to create our Temporal Workflow application.
+
+Within src/main/java/ create the "helloworld" directory and within src/main/java/helloworld create the following files:
+
+- HelloWorldActivity.java
+- HelloWorldActivityImpl.java
+- HelloWorldWorkflow.java
+- HelloWorldWorkflowImpl.java
+- HelloWorldStarter.java
+- HelloWorldWorker.java
+
+The following code snippets can be copied and pasted directly into your Java files. However, make sure that the first line of each file specifies the package name:
+
+```java
+package helloworld;
+```
+
+### Activity
+
+First, let's define our Activity. Activities are meant to handle non-deterministic code that could result in unexpected results or errors. But for this tutorial all we are doing is taking a string, appending it to "Hello", and returning it back to the Workflow.
+
+Open HelloWorldActivity.java and define the Activity interface:
+
+<!--SNIPSTART java-hello-world-sample-activity-interface-->
+<!--SNIPEND-->
+
+### Workflow
+
+A Workflow is a special function that organizes the sequence of Activity executions. Opposed to an Activity, a Workflow is deterministic in nature. It is where you can specify Activity options, such as what to do if the Activity results in an error or when to time out the Activity execution. Since our Activity is just formatting strings we will only specify the required timeout.
+
+Open HelloWorldWorkflow.java and define the Workflow interface:
+
+<!--SNIPSTART java-hello-world-sample-workflow-interface-->
+<!--SNIPEND-->
+
+Open HelloWorldWorkflowImpl.java and define the Workflow:
+
+<!--SNIPSTART java-hello-world-sample-workflow-->
+<!--SNIPEND-->
+
+### Starter
+
+There are two ways to start a Workflow, either via the CLI or via a starter program. In this tutorial we will start our Workflow by invoking it with a starter program.
+
+Open HelloWorldStarter.java and paste in the following code:
+
+<!--SNIPSTART java-hello-world-sample-workflow-starter-->
+<!--SNIPEND-->
+
+In the code above, we are creating a Temporal client and using it to host the Workflow with the Temporal server. Said another way, this starter program is sending the Workflow to the Temporal server where it will be broken into Tasks. The Tasks will be sent to the Task Queue that we supplied in the Workflow options.
+
+### Worker
+
+A Worker is the process that will actually execute the Workflow and Activity functions as Tasks.
+
+Open HelloWorldWorker.java and paste in the following code:
+
+<!--SNIPSTART java-hello-world-sample-worker-->
+<!--SNIPEND-->
+
+In this function we are creating a Temporal client and using it to create a new Worker. The Worker is registered to handle both HellowWorldWorkflow and HelloWorldActivity Tasks and is configured to listen to the same Task Queue that the Workflow and Activity Tasks are sent to.
```

Comment:
I think this sentence is so laden with terminology that it will be very hard for new users.